replications/internal/queue_management.go

```diff
@@ -136,62 +136,51 @@ func (rq *replicationQueue) run() {
 			// that rq.SendWrite will be called again in this situation and not leave data in the queue. Outside of this
 			// specific scenario, the buffer might result in an extra call to rq.SendWrite that will immediately return on
 			// EOF.
-			for rq.SendWrite() {
-			}
+			rq.SendWrite()
 		}
 	}
 }
 
 // SendWrite processes data enqueued into the durablequeue.Queue.
 // SendWrite is responsible for processing all data in the queue at the time of calling.
 // Network errors will be handled by the remote writer.
-func (rq *replicationQueue) SendWrite() bool {
+func (rq *replicationQueue) SendWrite() {
 	// Any error in creating the scanner should exit the loop in run()
 	// Either it is io.EOF indicating no data, or some other failure in making
 	// the Scanner object that we don't know how to handle.
 	scan, err := rq.queue.NewScanner()
 	if err != nil {
-		if err != io.EOF {
+		if !errors.Is(err, io.EOF) {
 			rq.logger.Error("Error creating replications queue scanner", zap.Error(err))
 		}
-		return false
+		return
 	}
 
 	for scan.Next() {
-		// An io.EOF error here indicates that there is no more data
-		// left to process, and is an expected error.
-		if scan.Err() == io.EOF {
-			break
-		}
-
-		// Any other here indicates a problem, so we log the error and
-		// drop the data with a call to scan.Advance() later.
-		if scan.Err() != nil {
+		if err := scan.Err(); err != nil {
+			if errors.Is(err, io.EOF) {
+				// An io.EOF error here indicates that there is no more data left to process, and is an expected error.
+				return
+			}
+			// Any other error here indicates a problem reading the data from the queue, so we log the error and drop the data
+			// with a call to scan.Advance() later.
 			rq.logger.Info("Segment read error.", zap.Error(scan.Err()))
-			// TODO: Add metrics collection for dropped data here.
```

Comment:
I am not 100% sure if it is possible to have `scan.Next()` return true and `scan.Err()` not be nil. I could not find a way to generate that scenario for testing. So at this point, I am leaving the error checking and logging in, but dropping the metrics collection. Adding a metrics collector here that is not able to be tested makes me nervous since they like to panic if the labels aren't set up correctly, and that eludes static analysis.